docs/Homebrew-brew-Maintainer-Guide.md

```diff
@@ -0,0 +1,102 @@
+
+This document describes a few components of the `Homebrew/brew` repository that are useful for maintainers to
+be aware of, but don't necessarily need to appear in documentation for most users and contributors.
+
+## Reviewing PRs
+
+Using `gh pr checkout NUMBER` is a super easy way to check out a PR branch using the GitHub CLI.
+
+When reviewing a PR, use "comment", "approve", or "request changes" when submitting based on the following guidelines:
+
+- Comment: if the PR isn't quite ready to be merged
+- Approve: if you feel that the PR is in a good state to be merged, even if there are
+  non-blocking changes you'd like to be made
+- Request changes: if you feel strongly that the PR is likely to cause a problem for users or
+  have a genuine reason to oppose the PR.
+
+## Merging PRs
+
+Merging should be done using the standard Merge button in the `Homebrew/brew` repository to preserve history and GPG commit signing. The Squash and Merge and Rebase and Merge buttons are disabled.
+
+PRs must meet the following conditions to be merged:
+
+- Have at least one maintainer (or `BrewTestBot`) approval. See [below](#automatic-approvals)
+  for more details about how `BrewTestBot` approves PRs.
+- Have passing CI. This is a _mandatory_ step. PRs with failing CI should _never_ be merged.
+  See [below](#ci) for more information about `Homebrew/brew` CI.
+
+If possible, PRs should also have:
+
+- Linear commit history (i.e. no merge commits in PR branches).
+- GPG-signed commits (see the private `ops` repository for instructions on setting this up).
+
+## Automatic approvals
+
+To ensure that each PR has the opportunity to be seen and reviewed by any other maintainers who wish
+to take a look, all PRs require an approval before they can be merged. However, not every PR is
+reviewed by another maintainer, and some PRs are urgent enough that they need to be merged without
+an approval by another maintainer.
+
+As a compromise between always needing a review and allowing maintainers to merge PRs they deem ready,
+the `Triage` CI job will ensure that PRs cannot be merged until they've been open for 24 hours
+(only counting hours that occur during the business week). After the triage period has expired, the
+CI job will show up as "passed" and `BrewTestBot` will approve the PR, allowing it to be merged.
+This gives all maintainers a reasonable opportunity to review every PR, but won't block any PR for lack
+of reviews.
+
+If the PR is urgent enough that it is necessary to bypass that 24 hour window, the `critical` label
+can be applied to the PR. When this label is applied, the `Triage` CI job will immediately be
+successful and `BrewTestBot` will approve the PR.
+
+## CI
+
+Every PR in `Homebrew/brew` runs a series of CI tests to try to prevent bugs from being introduced.
+**A PR _must_ have passing CI before it can be merged**.
+
+There are many checks that run on every PR. The following is a quick list of the various checks and what they represent:
```

Comment:
I'm somewhat tempted to either remove these or make it clear that this list isn't complete. Otherwise, it seems likely to bitrot.

Reply:
I don't feel super strongly either way. I do think there's some value in having them listed because it's not super straightforward to figure out what each one does (at least in my opinion as someone who still needs to dive into the code to figure it out every once in a while even though I'm around this repo a ton).

I wouldn't be opposed to removing some of the less-frequently-filing ones that arguably aren't as important to list (e.g. docker, vendored gems)

I can definitely add a "this list is non-exhaustive and can change over time" kind of thing that would hopefully prompt someone who realizes that the list is outdated to open a PR.

Reply:
> I wouldn't be opposed to removing some of the less-frequently-filing ones that arguably aren't as important to list (e.g. docker, vendored gems)
> 
> I can definitely add a "this list is non-exhaustive and can change over time" kind of thing that would hopefully prompt someone who realizes that the list is outdated to open a PR.

Yeh, this might make sense!